File: number_lines/FileRegexExample.php
<?php
$myfile = fopen("test.txt", "r") or die("Unable to open file!");
$test_str = fread($myfile,filesize("test.txt"));
fclose($myfile);
$re = '/^[0-9]+$/m';
preg_match_all($re, $test_str, $matches, PREG_SET_ORDER, 0);
var_dump($matches);
?>

File: number_lines/FileRegexExample.pl
open my $fh, '<', 'test.txt' or die "Can't open file $!";
read $fh, my $file_content, -s $fh;
close $fh;
my $regex = qr/^([0-9]+)$/mp;
my @matches = $file_content =~ /$regex/g;
print join(',', @matches);